### PR TITLE
Fixes #26219 - state control from Rec Logic

### PR DESCRIPTION
**Pull request by sjha4**
This PR is a continuation of https://github.com/Katello/katello/pull/8008 which is part of 3.11. 3.12 onwards, we are dropping the enabled field on the sync plan table and relying entirely on the foreman tasks recurring logic status. The steps to test are the same as in the above linked PR. We should see no change in functionality.

Since we are dropping the enabled field in 3.12, the katello:upgrades:3.9:migrate_sync_plans task is no longer relevant and can be removed.
_Note: Assumes 3.8 -> 3.12 upgrades are not supported._

### CODE REVIEW

**Comment by jturel** [comment]
[test katello]

re-testing since we had some rubocop updates

**Comment by jturel** [comment]
I think this can be removed:

https://github.com/Katello/katello/blob/ef0083b8d80ae4e0265d8db36ee1216fb3ebb140/db/seeds.d/111-upgrade_tasks.rb#L7

**Comment by jturel** [comment]
Interesting test failure as well, not sure if related

**Comment by sjha4** [comment]
Frozen time :smile: ..I'll take a look at the test..We should remove the db/seeds reference..Will push with the changes..

**Comment by sjha4** [comment]
The test error seems to have been random. Can't reproduce locally. Will have to wait for the jenkins to finish to confirm.

**Comment by jturel** [comment]
Toggling the sync plan or recurring logic updates both models which is the intention of the change

I noticed a problem when using hammer which is a separate bug:

hammer sync-plan update --id=4 --enabled=false --organization-id=2

```
2019-04-09T20:11:33 [I|app|994efa11] Started PUT "/katello/api/sync_plans/4" for ::1 at 2019-04-09 20:11:33 +0000                                                                                                 
2019-04-09T20:11:33 [I|app|994efa11] Processing by Katello::Api::V2::SyncPlansController#update as JSON                                                                                                           
2019-04-09T20:11:33 [I|app|994efa11]   Parameters: {"api_version"=>"v2", "id"=>"4", "sync_plan"=>{}}                                                                                                              
2019-04-09T20:11:33 [D|app|994efa11] Authenticated user admin against INTERNAL authentication source                                                                                                              
2019-04-09T20:11:33 [D|app|994efa11] Current user set to foreman_admin (admin)                                                                                                                                    
2019-04-09T20:11:33 [I|app|994efa11] Authorized user admin(Admin User)                                                                                                                                            
2019-04-09T20:11:33 [I|app|994efa11] Current user set to admin (admin)                           
2019-04-09T20:11:33 [D|app|994efa11] Current location set to none                                                                                                                                                 
2019-04-09T20:11:33 [D|app|994efa11] Current organization set to none                                                                                                                                             
2019-04-09T20:11:33 [D|app|994efa11] Current location set to none                                                                                                                                                 
2019-04-09T20:11:33 [D|app|994efa11] Current organization set to none                                                                                                                                             
2019-04-09T20:11:33 [E|app|994efa11] ActionController::ParameterMissing: param is missing or the value is empty: sync_plan                                                                                        
 | /home/vagrant/foreman/.vendor/ruby/2.5.0/gems/actionpack-5.2.1/lib/action_controller/metal/strong_parameters.rb:443:in `require'                                                                               
 | /home/vagrant/katello/app/controllers/katello/api/v2/sync_plans_controller.rb:126:in `sync_plan_params'                                                                                                        
 | /home/vagrant/katello/app/controllers/katello/api/v2/sync_plans_controller.rb:68:in `update'   
```

It worked when I specified by name:

```
hammer sync-plan update --name="mysyncplan-zoo" --enabled=false --organization-id=2
[Foreman] Password for admin:
Sync plan updated.
```

Pretty sure it's a bug - let me know and I can file an issue

File: app/controllers/katello/api/v2/sync_plans_controller.rb [hunk 1]
```diff
@@ -119,7 +122,8 @@ def find_plan
     end
 
     def sync_plan_params
-      params.require(:sync_plan).permit(:name, :description, :interval, :sync_date, :product_ids, :enabled, :cron_expression)
+      params[:sync_plan][:enabled] = params[:enabled] unless params[:enabled].nil?
+      sync_plan_param = params.require(:sync_plan).permit(:name, :description, :interval, :sync_date, :product_ids, :enabled, :cron_expression)
     end
```
Comment (sjha4) [inline app/controllers/katello/api/v2/sync_plans_controller.rb]: @jturel: Made this change here to add params[:sync_plan][:enabled] to sync_plan_params before the require validation. Removing the enabled field from the model meant that the param was no longer part of the param[:sync_plan].

**Comment by jturel** [comment]
Tested with recent change via hammer:

```
hammer sync-plan update --id=4 --enabled=false --organization-id=2                                                                            
[Foreman] Password for admin:
Sync plan updated.
```

ACK x 2

**Pull request merged**